kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -125,19 +133,29 @@ class ResidualBasedBlockBuilderAndSolver
         // Validate default parameters
         Parameters default_parameters = Parameters(R"(
         {
-            "name" : "ResidualBasedBlockBuilderAndSolver"
+            "name"                   : "ResidualBasedBlockBuilderAndSolver",
+            "scale_diagonal"         : false,

```

Comment:
you are mixing up different things here:

as i understand some of these parameters control the values used in the diagonal, while another is related to showing or not warnings.

1 - i would use a single parameter with multiple possible values to control the type of diagonal employed
    (to same something "average_diagonal", "max_diagonal", "min_diagonal" ... )
2 - the silencing of warning has nothing to do with this PR, so in my opinion it should not be here.

Reply:
ok, about keeping 2 here but keeping the current behaviour as default